### PR TITLE
sample oozie test job

### PR DESCRIPTION
**Pull request by ekund**
Added a tools directory where we can use small smoke-test tools.

Added a sample oozie test case that tests an oozie shell action and pushes the resulting job id to hdfs for verification.

I envision more simple tools being added as time goes on.

### CODE REVIEW

File: tools/oozie/oozie_simple_test/workflow.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      <shell xmlns="uri:oozie:shell-action:0.1">
+         <job-tracker>${jobTracker}</job-tracker>
+         <name-node>${nameNode}</name-node>
+         <configuration>
+            <property>
+               <name>mapred.job.queue.name</name>
+               <value>${queueName}</value>
+            </property>
+            <property>
+               <name>oozie.launcher.mapreduce.map.memory.mb</name>
```
Comment (pu239ppy) [inline tools/oozie/oozie_simple_test/workflow.xml]: There is already a default in oozie-site.xml it mirros same settings as mapred-site.xml, but prepends oozie.launcher
https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc-hadoop/templates/default/ooz_oozie-site.xml.erb#L357

https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc-hadoop/templates/default/ooz_oozie-site.xml.erb#L367

**Comment by http-418** [comment]
I really want to merge this, because it would be useful to me, but I don't understand whether Lev's comment should be addressed, first.

**Comment by ekund** [comment]
@pu239ppy - the overrides provide a way to override default values, allowing to test if the defaults are set properly or not, as I've seen it cause issues.

**Comment by saurabh313** [comment]
I think oozie coordinator jobs are more appropriate for recurrent actions. And since this just a test job, shouldn't we have oozie workflow  job ? In that case, we avoid the onus of updating start and end time each time we run the job.

**Comment by ekund** [comment]
I have replaced the job with an even simpler shell action that is just a workflow and has no overrides.  I have also added a setup script to make running it easier.

**Comment by amithkanand** [comment]
Can you please squash the commits?

**Comment by ekund** [comment]
I will, once we all agree that the code is good and ready to merge.

**Comment by ekund** [comment]
Commits have been squashed.

File: tools/oozie/oozie_simple_test/setup.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+sudo hdfs dfs -mkdir -p /tmp/oozie/shell-hadoopsh/
+sudo hdfs dfs -copyFromLocal job.properties /tmp/oozie/shell-hadoopsh/
+sudo hdfs dfs -copyFromLocal workflow.xml /tmp/oozie/shell-hadoopsh/
```
Comment (cbaenziger) [inline tools/oozie/oozie_simple_test/setup.sh]: This `sudo` is incorrect; I originally read this as `sudo -u hdfs` which would work to upload data as the HDFS user on machines which are running HDFS daemons such as namenodes.

File: tools/oozie/oozie_simple_test/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ - A start and end time set to today
+
+Then, create your test HDFS directory, and create 2 subdirectories underneath - lib and output.
+
+Finally, put the following files into that HDFS directory.
+ - coordinator.xml  
+ - logoozieid.sh  
+ - test.sh  
+ - workflow.xml
+
```
Comment (cbaenziger) [inline tools/oozie/oozie_simple_test/README.md]: Mention `kinit`'ing as a user first and creating a kerberos identity for one (e.g. `ubuntu`?) There's a pointer to doing that in our Wiki under [`Add Kerberos principals by using kadmin.local on bootstrap node`](https://github.com/bloomberg/chef-bach/wiki/Creating-a-Kerberized-cluster).

**Comment by cbaenziger** [comment]
Instead of using `/tmp` which will collide if multiple users try to run this, how about using `${wf:user()}` in the paths?

**Pull request merged**